### DISCUSSION TITLE
Remove all files and folders except a directory from a particular archive

### DISCUSSION THREAD

**Discussion by blueray453** [borgbackup/borg#8254] [Q&A] [answered]
I want to remove all files and folders except `/media/ismail/8TBRaid0/_BusinessAnalytics/` from archive `backup-2023-01-01`

I checked https://github.com/borgbackup/borg/issues/686 and the command I came up with is:

    borg recreate --patterns-from patterns.txt --dry-run  --list --verbose /path/to/repo::backup-2023-01-01

The `patterns.txt` file looks like:

```
P sh
R /media/ismail/8TBRaid0
R /media/ismail/SSDWorking
+ /media/ismail/8TBRaid0/_BusinessAnalytics
- *
```
Is it right or am i missing something.

**Comment by ThomasWaldmann** [comment]
Use `borg list REPO::ARCHIVE` to see what the paths look like, so you can make up matching patterns (don't use a leading slash in the include/exclude patterns).

Then dry-run recreate and you'll see whether it works. If it doesn't, also check the patterns doc page.

**Reply by blueray453** [reply]
The archive was created using:

```
P sh
R media/ismail/8TBRaid0
R media/ismail/SSDWorking
```

Other than the leading slash, does the `patterns.txt` file look right?

**Comment by ThomasWaldmann** [comment, marked as answer]
Ehrm, for the `R` (root paths), you need to give a valid path. Using a relative path there might work if the cwd is right (in your case cwd == `/`, I guess), but giving an absolute path there is less prone to break.

The patterns file might be right with the leading slash removed, just try it with --list --dry-run.

**Reply by blueray453** [reply]
Thank you very much.

**Reply by ThomasWaldmann** [reply]
Hmm, maybe you need to also match the tail of the paths, using `*` or `**` (depending on the pattern style).

Otherwise it might only keep the 1 directory object that matches the given path, but not any file inside that directory.

**Comment by blueray453** [comment]
Here is the pattern file:
```
P sh

R media/ismail/8TBRaid0
R media/ismail/SSDWorking
+ media/ismail/8TBRaid0/_BusinessAnalytics/**
- *
```
When i run:

    borg recreate --patterns-from /media/ismail/WDPurple8TB/patterns-clean.lst --dry-run --list /media/ismail/WDPurple8TB/8TBRaid0_Backup::SafeX &> files.txt

It gives me:

```
x media/ismail/8TBRaid0/_BusinessAnalysisRequirementGathering/
- media/ismail/8TBRaid0/_BusinessAnalytics/.../playlist.m3u
```

https://borgbackup.readthedocs.io/en/stable/usage/create.html says:

> 
> ‘-’ = dry run, item was not backed up
> 
> ‘x’ = excluded, item was not backed up

In the context of recreate what does x and - mean?

**Reply by ThomasWaldmann** [reply]
They mean exactly what the docs say. In other words:

you don't want to see `x` for items you did not intend to exclude - check that.

`-` (item was not backed up) is expected for included items because you did a dry-run (which does not really back up anything).

**Reply by blueray453** [reply]
It ran successfully. Thank you very much.
